mantidproject/mantid
# Workbench: Project Save saves all plot types and not just basic plots

**Description of work.**
A previously merged pull request of mine left out some necessary features. This Pull request adds them back in. It adds the ability to save and load colorfill plots and errorbar plots, fully supported. However it should also support all other forms of matplotlib saving and if failed the saving/loading with print a warning in the console and not error outright.

**To test:**
- Load Workbench,
- Load a workspace fit for using color plots i.e. MARI
- Create a colorfill plots
- Create a plot with errorbars
- Save the project (File->Save Project)
- Load the project (File->Load Project)
- It should pop up the two graphs again after loading back in the workspace.

- Close Workbench
- Load the project again
- The two graphs should appear again after loading back in the workspace

*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
